shared/configuration/src/airflow_shared/configuration/parser.py

```diff
@@ -0,0 +1,2309 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Shared configuration parser for Airflow distributions."""
+
+from __future__ import annotations
+
+import contextlib
+import datetime
+import functools
+import itertools
+import json
+import logging
+import multiprocessing
+import os
+import pathlib
+import re
+import shlex
+import stat
+import subprocess
+import sys
+import warnings
+from base64 import b64encode
+from collections.abc import Generator, Iterable
+from configparser import ConfigParser, NoOptionError, NoSectionError
+from contextlib import contextmanager
+from copy import deepcopy
+from io import StringIO
+from json.decoder import JSONDecodeError
+from re import Pattern
+from typing import IO, Any
+from urllib.parse import urlsplit
+
+from packaging.version import parse as parse_version
+from typing_extensions import overload
+
+# TODO: Fix import to be from shared secrets backend when we do that
+from airflow.secrets import DEFAULT_SECRETS_SEARCH_PATH
```

Comment:
We can't import directly from `airflow.*` in shared, yea?

Reply:
Absolutely correct, check the TODO attached above.

I have two options:
* Either hardcode the `DEFAULT_SECRETS_SEARCH_PATH` variable now with a TODO
* Or import from core with a TODO

because soon this import will go into a shared module too because SecretsBackend is going to come to secrets backend too